modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/TxStateStorageAbstractTest.java

```diff
@@ -150,10 +167,14 @@ public void testCas() throws Exception {
 
     @Test
     public void testScan() throws Exception {
-        try (TxStateTableStorage tableStorage = createStorage()) {
-            TxStateStorage storage0 = tableStorage.getOrCreateTxStateStorage(0);
-            TxStateStorage storage1 = tableStorage.getOrCreateTxStateStorage(1);
-            TxStateStorage storage2 = tableStorage.getOrCreateTxStateStorage(2);
+        try (
+                var storageHolder0 = new AutoCloser<>(tableStorage.getOrCreateTxStateStorage(0), TxStateStorage::close);
```

Comment:
Funny thing, you actually write more code than there were or even would be if there was no `AutoCloser`:

```
var storageHolder0 = tableStorage.getOrCreateTxStateStorage(0); 
storageHolder0.close();
```

And that's without calling `.value()` on all this holders...

Reply:
On the other hand, the code with try-with-resources is safer, it closes the storage.

Do you think it's not important in tests and should be replaced with simply calling `close()` in the end?

Reply:
It's hard to tell, but I think `try-with-resources` is used for the clarity of the code, not for security (because you have to close resources either way). And with AutoCloser we have to write more code with no profit. It's all my opinion of course

Reply:
`try-with-resources` provides some safeness guarantees, you cannot forget to close a resource used with `try-with-resources`, but you can if you don't use this idiom.

On the other hand, in tests it's not that critical to leak a resource, so I removed `try-with-resources` altogether.